lib/rom/sql/relation/reading.rb

```diff
@@ -1086,13 +1086,16 @@ def coerce_conditions(conditions)
         # @api private
         def __join__(type, other, join_cond = EMPTY_HASH, opts = EMPTY_HASH, &block)
           if other.is_a?(Symbol) || other.is_a?(ROM::Relation::Name)
-            if join_cond.equal?(EMPTY_HASH) && !block
-              assoc = associations[other]
+            assoc = associations.key?(other) && associations[other]
+
+            if join_cond.eql?(EMPTY_HASH) && !block
```

Comment:
```suggestion
            if (join_cond.nil? || join_cond.empty?) && !block
```
I'd rather do this. It'd allow `nil` or `{}` or `[]` as join_cond, which are all valid IMO.

I'm not sure why this was `.equal?`, since that does object equality (so even passing in `{}` would be false since it's a different though identical object).